clients/go/pkg/worker/jobStreamer.go

```diff
@@ -0,0 +1,115 @@
+// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package worker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/camunda/zeebe/clients/go/v8/pkg/commands"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/entities"
+)
+
+type jobStreamer struct {
+	request         commands.DispatchStreamJobsCommand
+	jobQueue        chan entities.Job
+	workerFinished  chan bool
+	closeSignal     chan struct{}
+	metrics         JobWorkerMetrics
+	shouldRetry     func(context.Context, error) bool
+	backoffSupplier BackoffSupplier
+	retryDelay      time.Duration
+
+	closedMutex sync.Mutex
```

Comment:
:thought_balloon: This is very Java like, with the 2 mutexes...honestly I'm not too sure about it. Generally I'm not too sure about the whole thing anyway. Not that it doesn't work, I'm sure it does, but about the general design =/